### ISSUE TITLE
Notification not sent immediately, when followed by a LISTEN on an unlistened channel

### ISSUE THREAD

**Issue by akarelas** [mojolicious/mojo-pg#77]
* Mojo::Pg version: 4.24
* Perl version: v5.32.1
* Operating system: Ubuntu Linux 20.04

### Steps to reproduce the behavior

**Long description of behavior:**
When sending a notification with `$pubsub->notify`, if this command is followed (in the same event loop iteration) by a `$pubsub->listen` on an un-listened channel, then the previously aforementioned notification is not delivered to its listeners, until a new notification happens (on a listened channel, that must not be however the channel the original notify occurred on).

This was a long description of the bug, to aid with your debugging efforts.

**tl;dr**
If NOTIFY is immediately followed by LISTEN on an un-listened channel, the notification is not delivered immediately.

The minimal case is really small:

```perl
use Mojo::Base -strict;

use Test::More;

use Mojo::Pg;
use Mojo::IOLoop;

plan skip_all => 'set TEST_ONLINE to enable this test' unless $ENV{TEST_ONLINE};

my $pg = Mojo::Pg->new($ENV{TEST_ONLINE});
my $pubsub = $pg->pubsub;
my $db = $pg->db;

my @received;
$pubsub->listen(ch1 => sub {
    my (undef, $payload) = @_;
    push @received, $payload;
});

$db->notify('ch1' => 'notification text');
$pubsub->listen('ch2', sub {}); # without this line, the test passes

Mojo::IOLoop->timer(0.5, sub {
    Mojo::IOLoop->stop_gracefully;
});

Mojo::IOLoop->start;

is_deeply \@received, ['notification text'], 'payload is delivered';

done_testing;
```

### Expected behavior
The above test should pass, indicating the notification has been received by the listener at ch1.

### Actual behavior
The above case does not pass.

I'm not 100% sure this is not a PostgreSQL bug, but it exhibits itself in Perl programs too.

**Comment by akarelas** [comment]
The same bug also occurs when you replace the word `listen` with `unlisten` in this line:

```perl
$pubsub->listen('ch2', sub {}); # without this line, the test passes
```

**Comment by akarelas** [comment]
This bug affects [BoardStreams](https://boardstreams.dev/)

**Comment by akarelas** [comment]
I logged all statements on PostgreSQL, to see whether the statements are passing through, and saw this in the logs:

```
2021-07-29 17:06:22.954 UTC [23776] dbuser@modules LOG:  statement: LISTEN "ch1"
2021-07-29 17:06:22.954 UTC [23776] dbuser@modules LOG:  statement: LISTEN "mojo.pubsub"
2021-07-29 17:06:22.955 UTC [23775] dbuser@modules LOG:  statement: NOTIFY "ch1", 'notification text'
2021-07-29 17:06:22.955 UTC [23776] dbuser@modules LOG:  statement: LISTEN "ch2"
```

It seems all statements are passing through (i.e. it's not a bug of DBD::Pg)

**Comment by akarelas** [comment]
I would like to make an update, because I did some more research: The second LISTEN, doesn't actually need to be a LISTEN command for the test to fail. Almost any blocking SQL command will do.

Such as SELECT * FROM "foo" (provided table foo exists)

**Comment by akarelas** [comment]
Fixed in version 4.26, so closing.
